.github/workflows/justtest.yaml

```diff
@@ -0,0 +1,699 @@
+#
+# If a test requires a special setup, it is likely not to be run
+# here. Instead of extra complication of this workflow, consider
+# creating a separate one with a needed setup.
+#
+# Keep all the logic within a single file
+# ---------------------------------------
+#
+# The testing steps should be simple enough to feel like a shell
+# script. Once we experience a need to hide some logic behind an
+# our own action or a reusable workflow, we likely
+# overcomplicate the logic and splitting it doesn't really help
+# with understanding.
+#
+# Also, don't make this action reusable, because it also means
+# splitting the logic: we inject some steps here that are needed
+# for the outside workflow.
+#
+# Deliberate complications
+# ========================
+#
+# Perform several builds in one workflow
+# --------------------------------------
+#
+# It seems that it doesn't complicate things too much.
+#
+# Split testing to jobs per test suite
+# ------------------------------------
+#
+# At first, it allows to restart a smaller piece of the workload
+# if an unstable test failed.
+#
+# Next, it allows to achieve larger parallelization level and
+# the feedback from CI is faster this way.
+#
+# Cache submodules build
+# ----------------------
+#
+# It is too attractive to have such a fast build. We use the
+# submodule hash to don't rebuild it each time.
+#
+# }}} Design choices
+#
+# TODO
+# ====
+#
+# This workflow is not ready to replace existing workflows that
+# perform the same build types. The reason is that
+# tarantool/luajit uses the existing workflows as reusable ones.
+
+name: justtest
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    # {{{ Build configurations
+
+    strategy:
+      matrix:
+        build-conf:
+          # TODO: Do we really need to run all the tests on the
+          # debug and release builds?
+          #
+          # First, there are tests that uses error injections,
+          # which are only available on the debug build.
+          #
+          # Second, the debug build has assertions enabled, so it
+          # can spot a problem that passes unseen on the release
+          # build.
+          #
+          # Third, the debug build has Lua's strict mode enabled
+          # by default.
+          #
+          # OTOH, the release build is what a customer typically
+          # uses, so it is the primary target. It feels
+          # counter-intuitive to verify the primary target with
+          # less tests that the additional targets.
+          #
+          # However, a release deployment job anyway performs the
+          # release build and it is different from this one in
+          # fact (at least the dependencies are linked statically
+          # and a specific building container is used). So, maybe,
+          # it is better to concentrate on developer needs in this
+          # workflow and reduce testing on the release build?
+          - debug
+          - release
```

Comment:
Do we really need to run all the tests on `debug` and `release` build configurations? I would appreciate opinions on it.

If you're curious, how it costs: not so much currently (maybe an extra minute of waiting for CI). However, it is due to the parallelization and we have some limits here (60 jobs in parallel on the team plan). So, if we're going to add more build types, running all the tests for all of them would make things worse.

Also, I measured it without contention between several pull requests. So, a picture in a busy hour can be different.

Please, check out my doubts in this regard: these are in the quoted TODO item.